### PR TITLE
fix: handle failed db connection on application startup

### LINKED ISSUE #876: add server healthcheck
Applying system upgrades to docker-ce restarts the docker daemon and all running containers. Normally this isn't an issue but for some reason the db <-> server container healthcheck does not register properly and the server dies with a basic db not ready message:

```
port-of-mars-server-1  | [06:56:19.691] FATAL (30): the database system is starting up
port-of-mars-server-1  |     filename: "/code/server/src/index.js"
port-of-mars-server-1  |     err: {                                                                                
port-of-mars-server-1  |       "type": "DatabaseError",
port-of-mars-server-1  |       "message": "the database system is starting up",
port-of-mars-server-1  |       "stack":   
port-of-mars-server-1  |           error: the database system is starting up
port-of-mars-server-1  |               at Parser.parseErrorMessage (/code/server/node_modules/pg-protocol/dist/parser.js:287:98)
port-of-mars-server-1  |               at Parser.handlePacket (/code/server/node_modules/pg-protocol/dist/parser.js:126:29)
port-of-mars-server-1  |               at Parser.parse (/code/server/node_modules/pg-protocol/dist/parser.js:39:38) 
port-of-mars-server-1  |               at Socket.<anonymous> (/code/server/node_modules/pg-protocol/dist/index.js:11:42)
port-of-mars-server-1  |               at Socket.emit (node:events:513:28)
port-of-mars-server-1  |               at Socket.emit (node:domain:489:12)
port-of-mars-server-1  |               at addChunk (node:internal/streams/readable:315:12)
port-of-mars-server-1  |               at readableAddChunk (node:internal/streams/readable:289:9)
port-of-mars-server-1  |               at Socket.Readable.push (node:internal/streams/readable:228:10)
port-of-mars-server-1  |               at TCP.onStreamRead (node:internal/stream_base_commons:190:23)         
port-of-mars-server-1  |       "length": 104,           
port-of-mars-server-1  |       "name": "error",
port-of-mars-server-1  |       "severity": "FATAL",
port-of-mars-server-1  |       "code": "57P03",                                                                    
port-of-mars-server-1  |       "file": "postmaster.c", 
port-of-mars-server-1  |       "line": "2329",
port-of-mars-server-1  |       "routine": "ProcessStartupPacket"
port-of-mars-server-1  |     }                  
```

Not sure why the containers aren't respecting their docker compose configuration when restarted but should investigate a bit more on staging. In the meantime, automated upgrades to production have been disabled so whenever we upgrade docker on the machine we'll need to manually double check that the services are up.

### PR DESCRIPTION
**Pull request by sgfost**
resolves #876

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -97,6 +97,7 @@
     "node-schedule": "^1.3.2",
     "nodemailer": "^6.6.0",
     "nodemailer-mailgun-transport": "^2.1.3",
+    "p-retry": "4",
```
Comment (alee) [inline server/package.json:100]: is the latest version 5 problematic or have other dependency issues?

https://www.npmjs.com/package/p-retry?activeTab=versions

Comment (sgfost) [reply server/package.json:100]: We'd need to switch to ESM in node. From the author: https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c

Seems like something we should do but I still need to look into it more

Comment (alee) [reply server/package.json:100]: Gotcha, thanks! Transitioning to ESM seems like something worth doing at some point but no need to hold this up.

**Review by alee — APPROVED** [review]

File: server/package.json (hunk 1 above)
Comment (sgfost) [reply server/package.json:100]: I've done some looking into it and it still seems like a bit of a dumpster fire that I wouldn't want to touch just yet. The previous versions of these packages that have gone pure ESM seem stable enough to still make up 98% of npm downloads so we have some time

**Pull request merged**